File: backend/app/schemas/management.py
from __future__ import annotations

from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field


class ManagementQueryCreate(BaseModel):
    question: str = Field(..., min_length=4, description="部门管理人员提出的公共服务分析问题")
    area: str = Field(default="中关村街道", description="分析区域")
    horizon_days: int = Field(default=14, ge=1, le=90, description="模拟和分析的时间范围")


class SimulatedAppeal(BaseModel):
    case_id: str
    created_at: datetime
    district: str
    community: str
    location_label: str
    latitude: float
    longitude: float
    category: str
    severity: Literal["low", "medium", "high"]
    population_group: str
    description: str
    requested_resource: str
    status: Literal["待受理", "已派单", "处理中", "已办结"]


class ManagementAgentStep(BaseModel):
    key: str
    title: str
    input_summary: str
    output_summary: str
    evidence: list[str]
    confidence: float


class TimeSeriesPoint(BaseModel):
    label: str
    count: int
    high_severity: int


class CategoryMetric(BaseModel):
    label: str
    count: int
    ratio: float


class RegionInsight(BaseModel):
    name: str
    count: int
    high_severity: int
    top_category: str
    recommended_action: str
    latitude: float
    longitude: float


class ManagementAnalysisResult(BaseModel):
    analysis_id: str
    created_at: datetime
    original_question: str
    normalized_question: str
    topic: str
    extracted_keywords: list[str]
    simulated_count: int
    records: list[SimulatedAppeal]
    agent_steps: list[ManagementAgentStep]
    time_series: list[TimeSeriesPoint]
    category_metrics: list[CategoryMetric]
    region_insights: list[RegionInsight]
    answer: str
    recommendations: list[str]

File: frontend/src/types/management.ts
export interface ManagementQueryCreate {
  question: string;
  area: string;
  horizon_days: number;
}

export interface SimulatedAppeal {
  case_id: string;
  created_at: string;
  district: string;
  community: string;
  location_label: string;
  latitude: number;
  longitude: number;
  category: string;
  severity: 'low' | 'medium' | 'high';
  population_group: string;
  description: string;
  requested_resource: string;
  status: '待受理' | '已派单' | '处理中' | '已办结';
}

export interface ManagementAgentStep {
  key: string;
  title: string;
  input_summary: string;
  output_summary: string;
  evidence: string[];
  confidence: number;
}

export interface TimeSeriesPoint {
  label: string;
  count: number;
  high_severity: number;
}

export interface CategoryMetric {
  label: string;
  count: number;
  ratio: number;
}

export interface RegionInsight {
  name: string;
  count: number;
  high_severity: number;
  top_category: string;
  recommended_action: string;
  latitude: number;
  longitude: number;
}

export interface ManagementAnalysisResult {
  analysis_id: string;
  created_at: string;
  original_question: string;
  normalized_question: string;
  topic: string;
  extracted_keywords: string[];
  simulated_count: number;
  records: SimulatedAppeal[];
  agent_steps: ManagementAgentStep[];
  time_series: TimeSeriesPoint[];
  category_metrics: CategoryMetric[];
  region_insights: RegionInsight[];
  answer: string;
  recommendations: string[];
}
